### PR TITLE
Change Ctrl-S-A to to select cell first, then whole document

### LINKED ISSUE #257: On the many absent virtues of "Select All"
In 5+ years of using qtconsole, I have never once used Ctrl+A/Select all. Except when I forget that it doesn't what I expect, which is to select the entire contents of the current cell.

Copying the whole buffer, with the banner and all seems like a very rare need indeed. Selecting an entire cell, on the other hand, is something I want to do several times an hour. 

Adding another "Select Cell" action is easy enough, but I wanted to suggest going further and having that take over <kbd>Ctrl-Shift-A</kbd> entirely. 

Is it just my minority opinion, or do the developers feel the same way? I'd be happy to submit the PR, it's a trivial change code-wise.

@ccordoba12 @takluyver @wmvanvliet

### PR DESCRIPTION
**Pull request by JeremySikes73**
closes #257

Hit once: select current cell
Hit again: select whole document
Hit yet again: select cell again

@ccordoba12 @takluyver @wmvanvliet

### CODE REVIEW

File: qtconsole/console_widget.py [hunk 1]
```diff
@@ -837,6 +837,24 @@ def _increase_font_size(self):
...
+    def select_all_smart(self):
+        """ Select current cell, or, if already selected, the whole document.
+        """
+        c = self._get_cursor()
+        sel_range = c.selectionStart(), c.selectionEnd()
+
+        c.clearSelection()
+        c.setPosition(self._get_prompt_cursor().position())
+        c.setPosition(self._get_end_pos(),
+                                 mode=QtGui.QTextCursor.KeepAnchor)
```
Comment (wmvanvliet) [inline qtconsole/console_widget.py]: Nitpick: could you align `mode=` with the opening bracket of `c.setPosition(`?

File: qtconsole/console_widget.py [hunk 2]
```diff
@@ -837,6 +837,24 @@ def _increase_font_size(self):
...
+        new_sel_Range = c.selectionStart(), c.selectionEnd()
```
Comment (wmvanvliet) [inline qtconsole/console_widget.py]: Weird capitalization. How about: `new_sel_range`?

File: qtconsole/console_widget.py [hunk 3]
```diff
@@ -837,6 +837,24 @@ def _increase_font_size(self):
...
+        if sel_range == new_sel_Range:
+            # cell already selected, expand selection to whole document
+            self.select_all()
```
Comment (wmvanvliet) [inline qtconsole/console_widget.py]: Let's rename `select_all` to `select_document` to avoid confusion.

File: qtconsole/console_widget.py (hunk 1 above)
Comment (JeremySikes73) [reply qtconsole/console_widget.py]: Fixed

File: qtconsole/console_widget.py (hunk 2 above)
Comment (JeremySikes73) [reply qtconsole/console_widget.py]: Fixed

File: qtconsole/console_widget.py (hunk 3 above)
Comment (JeremySikes73) [reply qtconsole/console_widget.py]: Fixed

**Comment by JeremySikes73** [comment]
@wmvanvliet , I've fixed all the issues in the review. Thanks.

**Comment by wmvanvliet** [comment]
Ok! +1 to merge from me.

**Pull request merged**